### DISCUSSION TITLE
Is the auto-generated entity ID globally unique?

### DISCUSSION THREAD

**Discussion by verigle** [milvus-io/milvus#3860] [Q&A and General discussion] [answered]
当我插入向量时， 使用自动创建向量时，自动生成的🆔id 是全局唯一的吗？

如果我想修改 id 对应的向量，是否提供了 update 的 api？

如果我先插入了一个向量， 此时id 自动创建，如果我想修改 某个id 对应的向量，能否通过id 删除向量，然后插入新向量时在指定这个id 呢？

**Comment by shiyu22** [comment, marked as answer]
1. When I insert a vector, is the auto-generated id globally unique?
Yes, it's unique.

2. Is there an API for updates if I want to change the vector corresponding to an id?
There is currently no interface to update the id, but it can be done by deleting the vector and inserting it.

3. When I insert a vector first, the id is automatically created. And I want to modify the vector for an id, can I delete the vector via the id, and then specify the id when I insert a new vector?
The update process as I answered above, but this operation can only be implemented on a collection of user-defined IDs. 
In other words, that ID in the same collection must be either user-generated or auto-generated. Can't be both. So this operation can't be implemented on a collection of auto-generated IDs.
